riak_test/tests/sibling_benchmark.erl

```diff
@@ -0,0 +1,222 @@
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% ---------------------------------------------------------------------
+
+-module(sibling_benchmark).
+
+%% @doc Microbenchmark of sibling convergence. Verifies the number of
+%% siblings are in the same magnitude with number of update
+%% concurrency. To test with previous version, add a configuration
+%% like this to your harness::
+%%
+%%      {sibling_benchmark,
+%%       [{write_concurrency, 2},
+%%        {duration_sec, 30},
+%%        {leave_and_join, 0}, %% number of cluster membership change during the benchmark
+%%        {version, previous}]
+%%        %%{version, current}]
+%%      },
+
+
+-export([confirm/0]).
+-include_lib("eunit/include/eunit.hrl").
+
+%% keys for non-multipart objects
+-define(TEST_BUCKET,        "riak-test-bucket").
+-define(KEY_SINGLE_BLOCK,   "riak_test_key1").
+
+%% keys for multipart uploaded objects
+-define(KEY_MP,        "riak_test_mp").  % single part, single block
+
+
+
+confirm() ->
+    RTConfig = rt_config:get(sibling_benchmark, []),
+    Concurrency = proplists:get_value(write_concurrency, RTConfig, 2),
+
+    CSConfig0 = rtcs:replace_cs_config(connection_pools,
+                                       [
+                                        {request_pool, {Concurrency*2, 0} },
+                                        {bucket_list_pool, {rtcs:bucket_list_pool_size(), 0} }
+                                       ],
+                                       rtcs:cs_config()),
+    CSConfig1 = rtcs:replace_cs_config(leeway_seconds,
+                                       5, CSConfig0),
+    CSConfig = rtcs:replace_cs_config(gc_interval,
+                                       10, CSConfig1),
+    ConnConfig = [{cs, CSConfig}],
+
+    {UserConfig, {RiakNodes, _CSNodes, _Stanchion}} =
+        case proplists:get_value(version, RTConfig, current) of
+            current -> rtcs:setup(4, ConnConfig);
+            previous ->
+                PrevConfig =  ConnConfig ++ rtcs:previous_configs(),
+                rtcs:setup(4, PrevConfig, previous)
+        end,
+
+    %% setting up the stage
+    ?assertEqual([{buckets, []}], erlcloud_s3:list_buckets(UserConfig)),
+    lager:info("creating bucket ~p", [?TEST_BUCKET]),
+    ?assertEqual(ok, erlcloud_s3:create_bucket(?TEST_BUCKET, UserConfig)),
+    %% The first object
+    erlcloud_s3:put_object(?TEST_BUCKET, ?KEY_SINGLE_BLOCK, <<"boom!">>, UserConfig),
+
+    get_counts(RiakNodes, ?TEST_BUCKET, ?KEY_SINGLE_BLOCK),
+    lager:info("====================== run benchmark ====================="),
+    {ok, Pid} = start_stats_checker(RiakNodes),
+
+    lager:info("write_concurrency: ~p", [Concurrency]),
+    Writers = [start_object_writer(UserConfig, 100) || _ <- lists:seq(1, Concurrency)],
```

Comment:
Is there any specific reason to adopt 100 msec interval between writes? (not intend to blame, just want to know)
Making this interval configurable may be nice?